ROX-33830: Update VM scanning docs to match current implementation (#112377)
openshift/openshift-docs: modules/recommended-requirements-vm-scanning.adoc

Reviewer: Above this, do you mind putting the required ID? It should be:

`[id="factors-affecting-throughput_{context}"]`

Author: Done, IDs added.

* Other workloads: While {product-title-short} processes VMs, {product-title-short} components keep processing the rest of the usual workloads. These components include Scanner V4, and therefore such workloads reduce the VM scanning throughput. 
+
To account for these factors and their variability, the suggested capacity for each deployment type includes a headroom of at least 100%. Therefore, even if the processing time doubles due to an increased number of packages or vulnerabilities, {product-title-short} can handle the specified number of VMs.

> == Compliance relay concurrency and buffer sizing

Reviewer: above, this, an ID is needed:

`[id="compliance-relay-concurrency_{context}"]`

Author: Done, IDs added.

The Compliance relay, which runs in the Compliance container of the Collector pod, manages the vsock connections from VMs and buffers index reports before forwarding them to Sensor. The following parameters affect relay capacity:

> * `ROX_VIRTUAL_MACHINES_MAX_CONCURRENT_VSOCK_CONNECTIONS`: Controls how many VMs can send index reports simultaneously to a single node. Changing this value is discouraged unless a single node handles many VMs and you observe performance problems in the Compliance container. Increasing this value increases the load on the Compliance container, and you must ensure that sufficient CPU and memory are allocated to the container.

Reviewer: ```suggestion
* `ROX_VIRTUAL_MACHINES_MAX_CONCURRENT_VSOCK_CONNECTIONS`: Controls how many VMs can send index reports simultaneously to a single node. Do not change this value unless a single node handles many VMs and you observe performance problems in the Compliance container. Increasing this value increases the load on the Compliance container, and you must ensure that sufficient CPU and memory are allocated to the container.
```

Author: Done

> * `ROX_VIRTUAL_MACHINES_INDEX_REPORTS_BUFFER_SIZE`: Controls the number of index reports buffered in Sensor before they are forwarded to Central. Increase this value only if Central is slow to process reports while Sensor is operating normally. Increasing this buffer causes Sensor to hold more index reports in memory, which can lead to Sensor running out of memory. Instead of increasing this value blindly, investigate the resource allocation for Central and Scanner V4 Matcher.

Reviewer: ```suggestion
* `ROX_VIRTUAL_MACHINES_INDEX_REPORTS_BUFFER_SIZE`: Controls the number of index reports buffered in Sensor before they are forwarded to Central. Increase this value only if Central is slow to process reports while Sensor is operating normally. Increasing this buffer causes Sensor to hold more index reports in memory, which can lead to Sensor running out of memory. Before changing this value, investigate the resource allocation for Central and Scanner V4 Matcher.
```

Author: Done

> * `ROX_VM_RELAY_MAX_REPORTS_PER_MINUTE`: Controls relay-side rate limiting for forwarding reports to Sensor per VM connection. Only increase this value if specific VMs are reporting infrequently (for example, once every one or two days instead of every four hours), which can indicate those VMs are being starved by the rate limiter. In that case, a modest increase (for example, from `1.0` to `2.0`) may help. Do not increase this value significantly unless the Scanner V4 Matcher has enough resources to handle the additional load.

Reviewer: ```suggestion
* `ROX_VM_RELAY_MAX_REPORTS_PER_MINUTE`: Controls relay-side rate limiting for forwarding reports to Sensor per VM connection. Only increase this value if specific VMs are reporting infrequently for example, once every one or two days instead of every four hours, which can indicate those VMs are being starved by the rate limiter. In that case, a modest increase, for example, from `1.0` to `2.0`, might help. Do not increase this value significantly unless the Scanner V4 Matcher has enough resources to handle the additional load.
```
IBM Style: no parentheses, use "might" instead of "may"

Author: Done: parentheses removed, "may" changed to "might".

> == Central enrichment throughput

Reviewer: Add ID above:
`[id="central-enrichment-throughput_{context}"]`

Author: Done, IDs added.
